testsuite/classlibrary/TestSerialPort.sc

```diff
@@ -0,0 +1,286 @@
+// Tests for SerialPort
+// Brian Heim, 22 May 2018
+//
+// These tests perform integration testing on SerialPort by creating an input/output pair of virtual serial ports
+// prior to each test.
+//
+// To run these tests on macOS/Linux, you need the `socat` utility. To run on Windows, you need to install `com0com`
+// and configure it for use with the CNCA0 and CNCB0 ports.
+
+TestSerialPort : UnitTest {
+
+	var input;
+	var output;
+
+	const kBufferSize = 8192;
+
+	setUp {
+		#input, output = this.createPorts();
+	}
+
+	tearDown {
+		this.destroyPorts();
+	}
+
+	// Create a pair of virtual serial ports and return their names
+	createPorts {
+		if(thisProcess.platform.name === \windows) {
+			^this.createCom0comPorts();
+		} {
+			^this.createSocatPorts();
+		};
+	}
+
+	destroyPorts {
+		if(thisProcess.platform.name === \windows) {
+			Error("Not yet implemented").throw;
+		} {
+			"killall socat".unixCmdGetStdOut();
+		};
+	}
+
+	createCom0comPorts {
+		var cmd = "\"C:/Porgram Files (x86)/com0com/setupc.exe install Portname=CNC0A Portname=CNC0B";
```

Comment:
`Porgram Files`?

Reply:
Sorry if this has been discussed elsewhere, and I haven't worked with Windows in quite some time, but:
Would it be possible to use Environment Variable here instead of a hardcoded path? Like ```%PROGRAMFILES(X86)%```

Reply:
@dyfer Yes, I will do that if I figure out a way to do automated serial port testing on Windows. Otherwise I am going to nuke this code branch anyway. :)